### PR TITLE
feat: add country flag to research username

### LINKED ISSUE #1406: Add country flag to research username
### Description

Everywhere on the platform we show username with country flag (comments/how-to) This was never integrated in the research module. That is an issue. Not cool.

### Page related

Will be used in :

- [Research page](https://community.preciousplastic.com/research/make-things-from-ocean-plastic) where it should come
- This layout already used in [how-to's ](https://community.preciousplastic.com/how-to/make-christmas-ornaments-from-upcycled-bottle-caps)
- Previous PR introducing the flag https://github.com/ONEARMY/community-platform/pull/1033

### Mockup
Add flag research

### PR DESCRIPTION
**Pull request by jojudre**
PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)
- [ ] - Latest `master` branch merged

PR Type

- [ ] New feature (non-breaking change which adds functionality)

## Description

add country flag to research username

## Git Issues

Closes #1406

## Screenshots/Videos

https://user-images.githubusercontent.com/42969746/160299344-8fc08324-3f92-43af-b598-2640f8073841.png

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on slack in the `platform-dev` channel.

### CODE REVIEW

**Comment by jojudre** [comment]
related PR that was closed - https://github.com/ONEARMY/community-platform/pull/1572

**Comment by jojudre** [comment]
@thisislawatts hi! i split logic to a helping function, did i do it right? can correct if needed

**Comment by davehakkens** [comment]
Looks good here!

**Review by thisislawatts — APPROVED** [review]

File: src/utils/getUserCountry.test.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  })
+
+  it('get user with country', () => {
+    user.location = null
+    user.country = 'uk'
+
+    expect(getUserCountry(user)).toBe('uk')
+  })
+
+})
```
Comment (thisislawatts) [inline src/utils/getUserCountry.test.ts:46]: Looks great @jojudre, thanks the time and effort introducing tests here!

**Pull request merged**